### PR TITLE
#244 avro test class

### PR DESCRIPTION
**Pull request by jgbriz**

### CODE REVIEW

File: src/main/java/net/coru/kloadgen/extractor/impl/SchemaExtractorImpl.java [hunk 1]
```diff
@@ -6,92 +6,58 @@
...
 
-  private final JsonExtractor jsonExtractor = new JsonExtractor();
+  private JsonExtractor jsonExtractor = new JsonExtractor();
 
-  private final ProtoBufExtractor protoBufExtractor = new ProtoBufExtractor();
+  private ProtoBufExtractor protoBufExtractor = new ProtoBufExtractor();
+
+  private JMeterHelper jMeterHelper = new JMeterHelper();
+
+  public SchemaExtractorImpl() {
```
Comment (jemacineiras) [inline src/main/java/net/coru/kloadgen/extractor/impl/SchemaExtractorImpl.java:47]: Is this needed?

File: src/test/java/net/coru/kloadgen/extractor/SchemaExtractorTest.java [hunk 2]
```diff
@@ -29,23 +39,36 @@
...
+
+  @Mock
+  private ProtoBufExtractor protoBufExtractor = Mockito.mock(ProtoBufExtractor.class);
+
+  @Mock
+  private JMeterHelper jMeterHelper = Mockito.mock(JMeterHelper.class);
 
   @BeforeEach
   public void setUp() {
+    schemaExtractor = new SchemaExtractorImpl(avroExtractor, jsonExtractor, protoBufExtractor, jMeterHelper);
```
Comment (jemacineiras) [inline src/test/java/net/coru/kloadgen/extractor/SchemaExtractorTest.java:70]: If you are using @InjectMocks, this line here is not needed.

File: src/test/java/net/coru/kloadgen/extractor/SchemaExtractorTest.java [hunk 3]
```diff
@@ -55,153 +78,72 @@ public void setUp() {
...
+
+    Mockito.verify(protoBufExtractor).processField(Mockito.any(TypeElement.class), Mockito.anyList(), Mockito.anyList(), eq(false));
+
   }
 
   @Test
-  @DisplayName("Should extract Optional Array")
-  void testFlatPropertiesOptionalArray() throws IOException {
+  @DisplayName("Test flatPropertiesList throws exception schema type not supported")
+  public void testFlatPropertiesListWithException() throws IOException, RestClientException {
```
Comment (jemacineiras) [inline src/test/java/net/coru/kloadgen/extractor/SchemaExtractorTest.java:133]: In JUnit 5 'public' modifier is not needed.

**Comment by Rivarsal** [comment]
Remember to fix this too! 

image

**Review by Rivarsal — APPROVED** [review]

**Review by fcampostato — APPROVED** [review]

**Review by jemacineiras — APPROVED** [review]

**Pull request merged**